### ISSUE TITLE
SW rendering frameskipping x1 cause black titlescreen and screen flickering (DarkStalkers/Vampire Chronicles)

### ISSUE THREAD

**Issue by Gamemulatorer** [hrydgard/ppsspp#12911]
### What happens?
On software rendering adding frame skipping to x1 or more can cause game hang/freeze tried and tested only to MH3rd HD (random freeze) Zenonia and Vampire Chronicle (during intro start)

### What *should* happen?
No hang or freeze when adding frame skipping on software rendering?

### What hardware, operating system, and PPSSPP version?  On desktop, GPU matters for graphical issues.

PPSSPP v1.9.3-808
Android 6.0.1
GLES 2.0
Mali 450

### screenrecorded video
[recording_20200516_235758.zip](https://github.com/hrydgard/ppsspp/files/4638702/recording_20200516_235758.zip)

### ge dump
[ge dump.zip](https://github.com/hrydgard/ppsspp/files/4638700/ge.dump.zip)

**Comment by Panderner** [comment]
For some reason even a Snapdragon 865 device can't handle most 3d games full speed in software renderer. Darkstalkers is forced by software renderer and it needs a high end device to run it full speed.

**Comment by Gamemulatorer** [comment]
> For some reason even a Snapdragon 865 device can't handle most 3d games full speed in software renderer. Darkstalkers is forced by software renderer and it needs a high end device to run it full speed.

My issue here is about frame skipping on software renderer not about performance 😕

**Comment by Gamemulatorer** [comment]
@Panderner FYI Darkstalker is playable on my phone I just need frame skipping x1 & auto frame skip enable. I got 25-30/60 fps 80-95% speed.

My only issue here is this game is hang/freeze during intro of the game as you can see in my screen recording video if frame skipping is set to x1 or more but during battle it didn't hang/freeze.

### screenshot
![Screenshot_20200517-024424](https://user-images.githubusercontent.com/37603562/82127882-5bb53d80-97e9-11ea-94bd-b0d330087556.png)

**Comment by unknownbrackets** [comment]
> For some reason even a Snapdragon 865 device can't handle most 3d games full speed in software renderer.

The software renderer doesn't use your GPU, it only uses your CPU.  This is known to be very slow - that's why devices have GPUs in the first place.  If it was mostly okay to do this rendering all on the CPU, cheaper phones would not have GPUs at all.

The software renderer struggles to get full speed on a 4.3 Ghz Intel CPU with 12 threads, because it only uses the CPU.

In contrast the puny Snapdragon 865 has one core at 2.84 Ghz, and the rest are slower.  Estimating using Ghz (this is not accurate), it's barely 1/3 as powerful as the Intel CPU at best.  Definitely too weak to render 3D games in the software renderer.

In other words: do not expect any mobile device to be able to run 3D games in the software renderer at any playable speed.  It may happen with VERY well optimized games.

-[Unknown]

**Comment by hrydgard** [comment]
Right, Darkstalkers only almost gets there on phones because I applied special optimization for the simple 2D rectangular operations that are the majority of what it draws. It is a very special case that just is not emulatable correctly without the software renderer.

**Comment by hrydgard** [comment]
Right so it doesn't seem that it hangs, it just doesn't draw the title screen. Maybe the frameskipping skipped something important. It can be bypassed by pressing the Start button. I don't think this is worth investigating further at this point ... frameskipping can break stuff.

**Comment by Gamemulatorer** [comment]
@hrydgard yes you're right it just blackscreen on title screen but I find out that enabling the render dupe to 60hz fixy this issue but when I enable it the screen is flickering.
### screen recording
[recording_20200518_083814.zip](https://github.com/hrydgard/ppsspp/files/4643338/recording_20200518_083814.zip)

**Comment by Gamemulatorer** [comment]
> @hrydgard yes you're right it's just blackscreen on title, but I find out that enabling the render dupe to 60hz fixed this issue, but when I enable the render dupe the screen is flickering.
> 
> ### screen recording
> [recording_20200518_083814.zip](https://github.com/hrydgard/ppsspp/files/4643338/recording_20200518_083814.zip)

@hrydgard @unknownbrackets any feedback? why the screen is flickering on SW renderer if I enable render duplicate frames to 60hz?

**Comment by hrydgard** [comment]
I have no idea, seems worth investigating but not top priority right now.

**Comment by Gamemulatorer** [comment]
I can also reproduce this issue on Snapdragon 430 Adreno 505 OpenGL 3.1+.
Black on title screen and screen flickering.

**Comment by Gamemulatorer** [comment]
I can't reproduce this issue again on latest git build using my Redmi Note 9. Closing..

**Comment by unknownbrackets** [comment]
Thanks for retesting.  I think this might be because of #13865.

-[Unknown]

**Comment by Gamemulatorer** [comment]
I try software rendering to Hokuto no Ken Raoh Gaiden
but using frame skipping x1 freeze the game.


https://user-images.githubusercontent.com/37603562/138047489-5b7da6fd-118b-446c-84e3-7c50545bb769.mp4

[software.ppdmp.zip](https://github.com/hrydgard/ppsspp/files/7379042/software.ppdmp.zip)
